httplink/__init__.py

```diff
@@ -1,248 +1,18 @@
-def parse_link_header(link: str) -> ParsedLinks:
-    """
-    Parse rfc8288 Link into a list of Link objects.
-
-    Raises ValueError if the link cannot be parsed.
-    """
-    result = []
-
-    # on first run, don't require a comma delimiter
-    matcher = RE_LINK_VALUE
-    while link:
-        # match leading <URI-Reference> and remove the matched string from `link`
-
-        uri_match = matcher.match(link)
-        if not uri_match:
-            break
-        link = link[uri_match.end():]
-
-        # on subsequent runs, DO require a comma delimiter
-        matcher = RE_LINK_VALUE_COMMA
-
-        uri = uri_match.group(1)
-
-        attributes = []
-        while True:
-            # match leading ;link-param and remove the matched string from `link`
-            param_match = RE_LINK_PARAM.match(link)
-            if not param_match:
-                break
-            link = link[param_match.end():]
-
-            # value is either a token or quoted
-            key, value_t, value_q = param_match.groups()
-            if value_t:
-                value = value_t
-            else:
-                value = unescape(value_q)
-            attributes.append((key, value))
-
-        result.append(Link(uri, attributes))
-
-    # Check for illegal leftovers
-    if not RE_TRAILING_COMMA.match(link):
-        raise ValueError('Bad link: {!r}'.format(link))
-
-    return ParsedLinks(result)
-
-
-# rfc8288 link_value (with optional leading ",")
-RE_LINK_VALUE = re.compile(r'''
-    ^
-    [\s,]*                                          # Skip empty elements (rfc7230#7).
-    <
-    ([^>]*)                                         # URI-Reference; contains no ">" according to grammar.
-    >
-''', re.X)
-
-# rfc8288 link_value (with mandatory leading ",")
-RE_LINK_VALUE_COMMA = re.compile(r'''
-    ^
-    \s*,[\s,]*                                      # Skip empty elements; require at least one delimiter (rfc7230#7).
-    <
-    ([^>]*)                                         # URI-Reference; contains no ">" according to grammar.
-    >
-''', re.X)
-
-# rfc8288 link-param (including ";" from link_value)
-RE_LINK_PARAM = re.compile(r'''
-    ^
-    \s*                                             # OWS
-    ;                                               # ";"
-    \s*                                             # OWS
-    ([-0-9A-Za-z!#$%&'*+-.^_`|~]+)                  # token
-    \s*                                             # BWS
-    =                                               # "="
-    \s*                                             # BWS
-    (?:
-        ([-0-9A-Za-z!#$%&'*+.^_`|~]+)               # token
-        |
-        "(                                          # quoted-string
-            (?:
-                [\t !#-[\]-~]                       # qdtext
-                |
-                \\[\t -~\x80-\xff]                  # quoted-pair
-            )*
-        )"
+from sys import version_info
+
+if version_info[0] == 2:
```

Comment:
I feel a bit unsure about this.
But on the other hand, Python2.7 tried to load the Python3 code and fails with a `SyntaxError`.